src/Components/CreateImageWizardV2/steps/TargetEnvironment/Gcp/index.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+
+import { Radio, Text, Form, Title, FormGroup } from '@patternfly/react-core';
+
+import { useAppDispatch, useAppSelector } from '../../../../../store/hooks';
+import {
+  changeGcpAccountType,
+  changeGcpEmail,
+  changeGcpShareMethod,
+  selectGcpAccountType,
+  selectGcpEmail,
+  selectGcpShareMethod,
+} from '../../../../../store/wizardSlice';
+import { ValidatedTextInput } from '../../../ValidatedTextInput';
+import { isGcpEmailValid } from '../../../validators';
+
+export type GcpShareMethod = 'withGoogle' | 'withInsights';
+export type GcpAccountTypes = 'google' | 'service' | 'group' | 'domain';
+
+const Gcp = () => {
+  const dispatch = useAppDispatch();
+
+  const accountType = useAppSelector((state) => selectGcpAccountType(state));
+  const shareMethod = useAppSelector((state) => selectGcpShareMethod(state));
+  const gcpEmail = useAppSelector((state) => selectGcpEmail(state));
+
+  return (
+    <Form>
+      <Title headingLevel="h2">
+        Target environment - Google Cloud Platform
+      </Title>
+      <Text>
+        Select how to share your image. The image you create can be used to
+        launch instances on GCP, regardless of which method you select.
+      </Text>
+      <FormGroup label="Select image sharing" isRequired>
+        <Radio
+          id="radio-with-description"
+          label="Share image with a Google acount"
+          name="radio-7"
+          description={
+            <Text>
+              Your image will be uploaded to GCP and shared with the account you
+              provide below.
+              <b>The image expires in 14 days.</b> To keep permanent access to
+              your image, copy it to your GCP project.
+            </Text>
+          }
+          isChecked={shareMethod === 'withGoogle'}
+          onChange={() => {
+            dispatch(changeGcpShareMethod('withGoogle'));
+          }}
+        />
+        <Radio
+          id="radio"
+          label="Share image with Red Hat Insights only"
+          name="radio-8"
+          description={
+            <Text>
+              Your image will be uploaded to GCP and shared with Red Hat
+              Insights.
+              <b> The image expires in 14 days.</b> You cannot access or
+              recreate this image in your GCP project.
+            </Text>
+          }
+          isChecked={shareMethod === 'withInsights'}
+          onChange={() => {
+            dispatch(changeGcpShareMethod('withInsights'));
+          }}
```

Comment:
`accountType` and principal should probably get reset here.

![image](https://github.com/RedHatInsights/image-builder-frontend/assets/49452678/fdbe6695-675f-44a4-be8b-6cb161e76143)

Reply:
So I think there are 3 options:
1. Don't reset them. In the submitHandler we use a conditional based on the shareMethod to generate the request, if the shareMethod is 'withInsights' we just ignore accountType and email. (lowest maintenance burden but also the least 'safe' solution)
2. Use the RTK listeners (consistent with how we are managing the architecture/release/target situation, but adds more maintenance burden).
3. Just set those to undefined when the shareMethod is set to 'withInsights'. (easy to understand and maintain, just not completely consistent with using RTK listeners we already have in place)

I don't have strong feelings either way, given the pros and cons of the above options which is your favorite?

Reply:
My vote would go to setting them to undefined so the store is not cluttered with things we won't use and go against the other choices.